server/src/main/java/io/druid/server/DruidNode.java

```diff
@@ -83,7 +83,7 @@ private void init(String serviceName, String host, Integer port)
 
     if(host == null && port == null) {
       host = getDefaultHost();
-      port = -1;
+      port = SocketUtil.findOpenPort(8080, host);
```

Comment:
If host and port are both null unit test will fail because port = -1 is not a valid setting.